test/pie-itemRadiusScale.html

```diff
@@ -0,0 +1,291 @@
+
+<html>
+<head>
+    <meta charset="utf-8">
+    <script src="lib/simpleRequire.js"></script>
+    <script src="lib/config.js"></script>
+    <style>
+        body {
+            margin: 0;
+        }
+
+        #main {
+            width:  100%;
+            height: 92%;
+        }
+
+        p {
+            text-align: center;
+        }
+
+        button {
+            padding:    5px 15px;
+            background: lightgray;
+            cursor:     pointer;
+        }
+
+        button:hover {
+            background: transparent;
+        }
+    </style>
+</head>
+<body>
+<p>
+    <button type="button"
+            onclick="selectRandomValues()">
+        Random selection
+    </button>
+</p>
+<div id="main"></div>
+<script>
+    let dataset = {
+        dimensions: ['key', 'value', 'selected'],
+        source: [
+            ['A', 1092, 112],
+            ['B', 768, 499],
+            ['C', 604, 257],
+            ['D', 238, 180]
+        ],
+    };
+
+    function selectRandomValues() {
+        dataset.source.forEach(row => {
+            row[2] = row[1] * Math.random();
+        });
+        chart.setOption({dataset});
+    }
+
+
+    function itemScale(params) {
+        const {data = []} = params;
+        return data[2] / data[1];
+    }
+
+    function itemScaleQuadratic(params) {
+        const {value, encode} = params;
+        const val = value[encode.value?.[0]];
+        const selected = value[encode.value?.[1]];
+        const scale = selected / val;
+        return Math.pow(scale, 1 / 2);
+    }
+
+    function itemScaleCubic(params) {
+        const {value, encode} = params;
+        const val = value[encode.value?.[0]];
+        const selected = value[encode.value?.[1]];
+        const scale = selected / val;
+        return Math.pow(scale, 1 / 3);
+    }
+
+    let chart;
+
+    require([
+        'echarts'
+    ], function (echarts) {
+        chart = echarts.init(document.getElementById('main'), null, {});
+
+        chart.setOption({
+            legend: {
+                itemStyle: {
+                    opacity: 1
+                }
+            },
+            tooltip: {trigger: 'item'},
+            dataset: dataset,
+            title: [
+                createTitle([1, 1], 'default'),
+                createTitle([1, 2], 'extra pie as background'),
+                createTitle([1, 3], 'donut style'),
+                createTitle([2, 1], 'cubic scale'),
+                createTitle([2, 2], 'quadratic scale'),
+                createTitle([2, 3], 'quadratic scale + donut style'),
+            ],
+            series: [
+                /** example #1: scaling based on a simple function */
+                createScaledSeries(
+                    [1, 1],
+                    {},
+                    itemScale
+                ),
+
+                /** example #2: add pie with the same angles,
+                 * but full radius and semitransparent background */
+                createBackgroundSeries([1, 2]),
```

Comment:
I would suggest providing two new options `series-pie.backgroundStyle` and `series-pie.backgroundStyle` like [series-bar.showBackground](https://echarts.apache.org/en/option.html#series-bar.showBackground) and [series-bar.backgroundStyle](https://echarts.apache.org/en/option.html#series-bar.backgroundStyle), where `series-pie.backgroundStyle.color` can be a callback function so that developers can set the background according to the piece color.

Reply:
You make make a seperate PR (or just in this if you like) for the background related feature because it also benifits the roseType pies.

Reply:
Intuitively, it seems that the background style in the series root will be applied to the entire pie. it's not obvious that the function is called on every element. If I'm trying to intuitively understand property names, then it seems more logical for me to put the style of the elements into itemStyle. Like `series-pie.itemStyle.backgroundStyle.color`.

But this entry `series-pie.backgroundStyle.opacity` with a numeric type rather than a function seems clear to me. This tells me that transparency will work the same on all elements

Reply:
Yes, I would slighter prefer to callback than under `itemStyle` because users don't have to write multiple times. Another option could be, `series-pie.backgroundStyle.color` has a default value `'auto'` meaning use color of each piece. [series-pie.colorBy](https://echarts.apache.org/en/option.html#series-pie.colorBy) is `'series'` by default, so it's intuitively OK for the background to use the color of each piece.